detection_rules/utils.py

```diff
@@ -486,6 +487,27 @@ def decorator(f: FuncT) -> FuncT:
     return decorator
 
 
+def load_toml_file(file_path: Path) -> dict[str, str]:
+    """Loads a TOML file and returns its contents as a dictionary."""
+    try:
+        with file_path.open("rb") as f:
+            return tomllib.load(f)
+    except tomllib.TOMLDecodeError as e:
+        raise ValueError(f"Failed to parse TOML file: {file_path}") from e
+
+
+def load_toml_rule_paths_by_id(file_paths: list[Path]) -> dict[str, Path]:
+    """Lightweight loader for raw TOML rule files. No Validation is done."""
+    rule_map: dict[str, Path] = {}
+    for directory in file_paths:
+        for toml_file in Path(directory).glob("**/*.toml"):
+            toml_data: dict[str, Any] = load_toml_file(toml_file)
+            rule_id = toml_data.get("rule", {}).get("rule_id")
+            if rule_id:
+                rule_map[rule_id] = toml_file
+    return rule_map
```

Comment:
What about using `RawRuleCollection`?

Reply:
Will take a look at using `RawRuleCollection`. It still contains some validation overhead, but probably will be minimal enough to not warrant having this separate function. I expect to switch over to this `RawRuleCollection`.

Reply:
Updated/Fixed in https://github.com/elastic/detection-rules/pull/4915/commits/500152b8cf11afb5e8702073f59ec5336524766f